pkg/plugins/golang/v3/scaffolds/internal/templates/main.go

```diff
@@ -239,12 +239,13 @@ func main() {
 
 {{ if not .ComponentConfig }}
```

Comment:
What about we add an e2e test with the leader election enabled for go/v3 and go/v4 scaffolded projects in: https://github.com/kubernetes-sigs/kubebuilder/tree/master/test/e2e?

In this case, I think we would need to create a NEW test which will update the config/manager to pass the arg with the value == true.

Reply:
Checked code in https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/common/kustomize/v2-alpha/scaffolds/internal/templates/config/manager/config.go#L118-L121

The e2e test in https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/generate_test.go#L38 and https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v4/generate_test.go#L38, the `--component-config` flag is not set, then will generate config/manager with `--leader-elect` set true.

So maybe it is enough to test the leader election enabled.
PTAL thanks.

Reply:
See that we have a test with the componentConfig enabled, see: https://github.com/kubernetes-sigs/kubebuilder/blob/cf154d6770ae19e65fbf3a7a2832b1e9edb3374a/test/e2e/v4/plugin_cluster_test.go#L91-L96

Then, we could add +1 test to mock the same but update the config/manager.yaml to ensure that the leader election is also enabled and test it with the namespace.

My concern here is that the namespace has not been added and we do not face any issues when we are using deploy IMG command to check it against a cluster. The error is only faced using make run to test it outside of the cluster. So that we need to ensure that the change will not break the prod/current behaviour. So we might want to address the need via doc and not in the default scaffold.

Reply:
Yes, agree with the concerning here.
But currently it has already included the `--leader-elect=true` with this test case:
https://github.com/kubernetes-sigs/kubebuilder/blob/cf154d6770ae19e65fbf3a7a2832b1e9edb3374a/test/e2e/v4/plugin_cluster_test.go#L84-L89

With the above code snippit will set `--component-config` to false (default).

Then will set `--leader-elect` flag to true:
https://github.com/kubernetes-sigs/kubebuilder/blob/5d59e40a7b7a2f35476a7b657973f63ad734bdb8/pkg/plugins/common/kustomize/v2-alpha/scaffolds/internal/templates/config/manager/config.go#L118-L121

PTAL thanks.

Reply:
Hi @astraw99, 

That is great. We have the desire for too long and many issues tracked for we add in a FAQ section.
So, wdyt about create a PR to add a FAQ section in the KB docs that would be like https://sdk.operatorframework.io/docs/faqs/ to address this one?

See all issues to track in this FAQ section that we need to add to the docs:  https://github.com/kubernetes-sigs/kubebuilder/issues?q=is%3Aissue+is%3Aopen+FAq

Reply:
OK, I will create a FAQ in the KB docs, and include some of the representational reported issues.
And update it continuously, to help more users :)

Reply:
@camilamacedo86 Opened a FAQ PR https://github.com/kubernetes-sigs/kubebuilder/pull/3044
PTAL thanks.